packages/workspace/src/common/workspace-protocol.ts

```diff
@@ -13,6 +13,11 @@ export const workspacePath = '/services/workspace';
 export const WorkspaceServer = Symbol('WorkspaceServer');
 export interface WorkspaceServer {
 
+    /**
+     * Returns the random identifier to the running theia instance
+     */
+    getDomain(): Promise<string | undefined>;
```

Comment:
There is no need to ask the server for a domain. As we are already using a domain to connect to it.
Why not `window.location`?
Also, the Workspace doesn't have anything to do with domains.

Reply:
`window.location` includes everything, such as the query string (`?a=b`).
This is the reason why electron was broken before, because it adds `?port=random-number`.
You shouldn't need more than `pathname` because the local storage already segregates based on the domain/subdomain/port.